tsl/src/nodes/vector_agg/function/minmax_templates.c

```diff
@@ -49,11 +49,13 @@ minmax_emit(void *agg_state, Datum *out_result, bool *out_isnull)
  * Templated parts for vectorized min(), max().
  */
 #define AGG_NAME MIN
-#define PREDICATE(CURRENT, NEW) ((CURRENT) > (NEW))
+#define PREDICATE(CURRENT, NEW)                                                                    \
+	(!isnan((double) NEW) && (isnan((double) CURRENT) || (CURRENT) > (NEW)))
```

Comment:
```suggestion
	unlikely(!isnan((double) NEW) && (isnan((double) CURRENT) || (CURRENT) > (NEW)))
```

Might be slightly better this way, since we mostly don't work with NaNs, although I doubt it does anything